torch/_dynamo/eval_frame.py

```diff
@@ -668,6 +668,8 @@ def toy_example(a, b):
     torch._C._log_api_usage_once("torch._dynamo.optimize")
     if disable or os.environ.get("TORCHDYNAMO_DISABLE", "") == "1":
         return _NullDecorator()
+    elif config.compiled_autograd_enabled:
+        torch._dynamo.compiled_autograd._initialize()
```

Comment:
Wont installing compiled autograd globally like this has bad global side effects?

If I have two models, one compiled and one eager -- then this will enable compiled autograd for the unrelated model.

Reply:
I don't think it should? I changed the autograd engine routing logic in engine.cpp:1270. it now needs
- a compiled being set (by _initialize)
- either
  - an autograd graph containing a CompiledFunctionBackward marked by compiled autograd
  - .backward called within the ctx manager override

the eager model's output on which we call .backward shouldn't have an autograd graph containing CompiledFunctionBackward, so autograd engine would route to eager.